packages/flutter/test/cupertino/text_field_test.dart

```diff
@@ -94,8 +92,8 @@ class PathPointsMatcher extends Matcher {
 
   @override
   bool matches(covariant Path item, Map<dynamic, dynamic> matchState) {
-    final Offset notIncluded = includes.firstWhere((Offset offset) => !item.contains(offset), orElse: () => null);
-    final Offset notExcluded = excludes.firstWhere(item.contains, orElse: () => null);
+    final Offset? notIncluded = includes.cast<Offset?>().firstWhere((Offset? offset) => !item.contains(offset!), orElse: () => null);
```

Comment:
Why is the `cast` needed? Seems like you should be able to do:
```suggestion
    final Offset? notIncluded = includes.firstWhere((Offset offset) => !item.contains(offset), orElse: () => null);
```
But I guess `firstWhere` is templated on the type of `includes` and probably doesn't allow null from the `orElse`.  Interesting.

Reply:
> But I guess firstWhere is templated on the type of includes and probably doesn't allow null from the orElse. Interesting.

Yes, this is unfortunately the reason for the cast...